src/hooks/index.ts

```diff
@@ -0,0 +1,9 @@
+import { useEffect, useRef } from "react";
+
+export const usePrevious = <T>(state: T): T | undefined => {
+  const ref = useRef<T>();
+  useEffect(() => {
+    ref.current = state;
+  });
```

Comment:
i believe this should have a `[]` as it's second param, otherwise it will perform this action on every re-render of its invoker